### DISCUSSION TITLE
"useLocalHTML" ??

### DISCUSSION THREAD

**Discussion by pairohit305** [LonelyCpp/react-native-youtube-iframe#125] [Q&A] [answered]
What is use of useLocalHTML will it boost performance or .... I am not getting the use of it ?

**Comment by LonelyCpp** [comment, marked as answer]
There are two ways to render the youtube player in the webview.

1. generate HTML based on the props and give it to webview as raw HTML string ([code](https://github.com/LonelyCpp/react-native-youtube-iframe/blob/651cdf5d2144f94e0f778acd2e7b30ffb7aaa85d/src/PlayerScripts.js#L136))
    * advantages: no need to load an extra webpage.
    * disadvantages: page will not have a domain name leading to some videos being "unavailable"
  
2. host the HTML on a remote source (github pages by default) and pass information with query parameters ([code](https://github.com/LonelyCpp/react-native-youtube-iframe/blob/master/iframe.html))
    * advantages: has a trusted domain name, which prevents all "unavailable" errors.
    * disadvantages: extra [3kb](https://github.com/LonelyCpp/react-native-youtube-iframe/blob/master/website/static/iframe.html) page load, github source could be unavailable (unlikely)

**Reply by pairohit305** [reply]
so basically useLocalHTML is the 1st one "generate HTML based on the props and give it to webview as raw HTML string" 
in that you said  some videos being "unavailable" ... so which type of videos are unavailable ?? one which the video owner set no
embed? if that case then no issues

**Reply by LonelyCpp** [reply]
> one which the video owner set no embed

This was my first guess as well. But there's no way to be certain. 

Videos were playing just fine on normal web embeds but failing to load here.
